hdkeychain/extendedkey_test.go

```diff
@@ -1013,3 +1014,38 @@ func TestZero(t *testing.T) {
 		}
 	}
 }
+
+func TestExtendedKey_MaximumDepth(t *testing.T) {
```

Comment:
Please just call this `TestMaximumDepth`.  Using snake case is discouraged by Effective Go guidelines.

Also, it's missing a comment to be consistent with the rest of the test functions.  The following should do if you don't already have something in mind:

```Go
// TestMaximumDepth ensures that attempting to retrieve a child key when already
// at the maximum depth is not allowed.
```